src/field/binary_towers/extension.rs

```diff
@@ -0,0 +1,344 @@
+//! contains implementation of binary extension fields using tower field arithmetic in multilinear basis as defined in [DP23b](https://eprint.iacr.org/2023/1784.pdf)
+use std::{
+  cmp::Ordering,
+  iter::{Product, Sum},
+  ops::{Add, AddAssign, Div, DivAssign, Mul, MulAssign, Neg, Rem, Sub, SubAssign},
+};
+
+use rand::{
+  distributions::{Distribution, Standard},
+  Rng,
+};
+
+use super::BinaryField;
+use crate::field::FiniteField;
+
+/// Binary extension field GF_{2^{2^K}} using binary towers arithmetic as explained in Section 2.3 of [DP23b](https://eprint.iacr.org/2023/1784.pdf)
+/// represented as vector of 2^K [`BinaryField`] components in multilinear basis,
+/// i.e. an element $b_v = \prod_{i=0}^{k-1}(v_i.X_i+(1-v_i))$ where $v_i$ is in GF(2).
+#[derive(Debug, Clone, Copy, Hash, PartialEq, Eq, PartialOrd, Ord)]
+pub struct BinaryFieldExtension<const K: usize>
+where [(); 1 << K]: {
+  /// coefficient of field element
+  pub coefficients: [BinaryField; 1 << K],
+}
+
+impl<const K: usize> BinaryFieldExtension<K>
```

Comment:
Should I be thinking of `BinaryFieldExtension<K>` as the binary tower with height `K`?

Reply:
Ah nice. Probably a nit, but what do you think of naming this `BinaryTower` so we can think of `GaloisField<K, 2>` as degree `K` extensions over `PrimeField<2>`?

Let it be known that it doesn't bother me either way, you should choose what you think is proper :)